File: pset6/sentimental-readability/readability.py
import string


def count_letters(text: str):
    letters = 0
    for character in text:
        if character in string.ascii_letters:
            letters += 1
    return letters


def count_words(text: str):
    words = 1
    for item in text:
        if item == " ":
            words += 1
    return words


def count_sentences(text: str):
    sentences = 0
    for item in text:
        if item in ['.', '?', '!']:
            sentences += 1
    return sentences


text = input("Text: ")

letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)

index = 0.0588 * letters * 100 / words - 0.296 * sentences / words * 100 - 15.8
index = round(index)

if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);

int count_words(string text);

int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    double index = 0.0588 * letters * 100 / words - 0.296 * sentences / words * 100 - 15.8;
    int rounded_index = (int) round(index);

    if (rounded_index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (rounded_index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", rounded_index);
    }
}

int count_letters(string text)
{
    int letters = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isupper(text[i]) || islower(text[i]))
        {
            letters += 1;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            words += 1;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}
